### PR TITLE
Daily 19 01 29

### PR DESCRIPTION
**Pull request by elaine1234**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: src/global.js [hunk 1]
```diff
@@ -9,13 +9,15 @@ module.exports = {
   nodeContainerClassName: 'node-container',
   edgeContainerClassName: 'edge-container',
   defaultNode: {
```
Comment (leungwensen) [inline src/global.js]: graph 级别的默认 shape type

File: src/graph/controller/event.js [hunk 2]
```diff
@@ -57,6 +59,21 @@ class Event {
     window && extendEvents.push(Util.addEventListener(window, 'keyup', originHandler));
   }
   _onCanvasEvents(e) {
+    const self = this;
+    const type = e.type;
+    if (type === 'click') {
+      clearTimeout(_timer);
+      _timer = setTimeout(() => {
+        self._triggerCanvasEvents(e);
+      }, 200);
```
Comment (leungwensen) [inline src/graph/controller/event.js]: 改成常量或者配置项？

File: src/graph/controller/item.js [hunk 3]
```diff
@@ -138,11 +138,15 @@ class ItemController {
     }
     graph.emit('beforeitemvisibilitychange', { item, visible });
     item.changeVisibility(visible);
-    // 若隐藏节点，则将与之关联的边也隐藏
     if (item.getType() === NODE) {
       const autoPaint = graph.get('autoPaint');
       graph.setAutoPaint(false);
       Util.each(item.getEdges(), edge => {
+        // 若隐藏节点，则将与之关联的边也隐藏
+        // 若显示节点，需要边两端的节点都是显示的
```
Comment (leungwensen) [inline src/graph/controller/item.js]: FIXME

File: src/graph/graph.js [hunk 4]
```diff
@@ -3,8 +3,8 @@
  * @author huangtonger@aliyun.com
  */
 
-const EventEmitter = require('@antv/g/lib/').EventEmitter;
 const G = require('@antv/g/lib');
+const EventEmitter = G.EventEmitter;
```
Comment (leungwensen) [inline src/graph/graph.js]: import G { EventEmitter } from '@antv/g';

File: src/global.js (hunk 1 above)
Comment (elaine1234) [reply src/global.js]: 9a5f2754c1f439a4532d4ac3c3eade64ae57d863

**Pull request merged**